lib/iris/_merge.py

```diff
@@ -1235,7 +1235,7 @@ def merge(self, unique=True):
                         if ma.is_masked(data):
                             data = array_masked_to_nans(data)
                         data = data.data
-                    data = da.from_array(data, chunks=data.shape)
+                    data = as_lazy_data(data, chunks=data.shape)
```

Comment:
The lines doing the mask->nans above this can now be removed as it's being done in `as_lazy_data`